### ISSUE TITLE
Code quality needs to be maintained

### ISSUE THREAD

**Issue by InfiniteCoder** [OpenGenus/cosmos#3128]
**This is a(n):**

- [ ] New algorithm
- [ ] Update to an existing algorithm
- [ ] Error
- [X] Proposal to the Repository

**Details:**

There are many programs here which have very poor quality. Programs that use single character variables, those literally abusing the language syntax to make code shorter(and unreadable) and programs using legacy tools/code. In this form, they are unhelpful to anyone. At one point, this repository will become a pile of useless codes, if these codes are not dealt with properly.

1. Really bad code needs to be removed
2. Salvageable code needs to be improved
3. All codes need to follow same guidelines
4. Whatever tools from GitHub marketplace that are applicable here need to be used. Especially those that ensure code quality.

I think it might even be a good idea to pause accepting new codes for a while until the existing are fixed. Then set up strict guidelines and a proper workflow for accepting submissions, and accept only those codes which are good.

Otherwise, all this effort might go in vain.

**Comment by the-saif-ahmad** [comment]
Not only do we need to fix the existing code, we need to create style guides for all languages to prevent it from happening again. Then even if someone does submit bad code, they can't use the lack of a style guide as an excuse.

**Comment by AdiChat** [comment]
Yes, this is an issue that needs to be attended immediately and the solutions by both of you look promising. 
You must share this at [our new forum](https://discourse.opengenus.org/) where it is more accessible and continue discussing there. 

We must arrive at definite measures to implement as soon as possible. 

**Let us make Cosmos great again.** 💪

**Comment by InfiniteCoder** [comment]
Created a post here https://discourse.opengenus.org/t/fixing-code-quality/47

**Comment by amirags** [comment]
Happy to contribute to this cause!

**Comment by AdiChat** [comment]
That is great @amirags 😃 Go ahead and solve issues one by one 👍 
You can, also, suggest and implement new solutions.

**Comment by amirags** [comment]
Sorry, lying low these days (blame the weather!). Planning to make a list of todos and take it forward. Will add further updates here.

**Comment by AdiChat** [comment]
No problem 👍 🗻  Stay safe and warm 🔥 
Sure. We are looking forward towards your plan. 😃

**Comment by amirags** [comment]
Happy New Year all!

Right! Started having a look at the code, guidelines etc. Adding some points to start with (let me know if discourse is better place for this):

a. Encourage adding/updating READMEs with each category/sub-category of algorithms. A brief definition should be encouraged to help in general understanding of given algorithm.
b. Encourage consistent use of filenames, folder names as well as variables, method names, comments within the code. Can be checked while reviewing PRs. 
c. Add content to links leading to 404 currently, e.g style guides
d. Use testing frameworks to add tests ? This will avoid multiple main issue.

**Comment by AdiChat** [comment]
Happy new year @amirags 🎉 

We need to prepare an idealist so, that we are able to track upcoming tasks/ ideas.  Ideally, Discourse is for open discussion and brainstorming. That being said, use the platform whether you feel comfortable. 👍 

I, strongly, agree with your conclusions. Adding the description of algorithms is a priority. Having consistent naming and covering all style guides are important as well.
Developing a testing framework for Cosmos is essential as this will form a basis for other sister projects. 

Together we can and we will achieve this 💪

**Comment by Divya063** [comment]
I think maybe implementing a code quality tool like [code climate](https://codeclimate.com/) will help.
